flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -147,9 +228,31 @@ public void run() {
 			}
 		} catch (Throwable t) {
 			fetcherRef.stopWithError(t);
+		} finally {
+			if (this.streamInfoList != null) {
+				try {
+					deregisterStreamConsumer();
+				} catch (Throwable t) {
+					fetcherRef.stopWithError(t);
+				}
+			}
 		}
 	}
 
+	private void deregisterStreamConsumer() throws ExecutionException, InterruptedException {
```

Comment:
@xiaolong-sn this needs to move to `KinesisDataFetcher::shutdownFetcher`

Reply:
Moved.